include/ensmallen_bits/age_moea/agemoea.hpp

```diff
@@ -0,0 +1,489 @@
+/**
+ * @file agemoea.hpp
+ * @author Satyam Shukla
+ *
+ * AGE-MOEA is a multi-objective optimization algorithm, widely used in
+ * many real-world applications. AGE-MOEA generates offsprings using
+ * crossover and mutation and then selects the next generation according
+ * to non-dominated-sorting and survival score comparison.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef ENSMALLEN_AGEMOEA_AGEMOEA_HPP
+#define ENSMALLEN_AGEMOEA_AGEMOEA_HPP
+
+namespace ens {
+
+/**
+ * This class implements the AGEMOEA algorithm.
+ *
+ * The algorithm works by generating a candidate population from a fixed
+ * starting point. At each stage of optimization, a new population of children
+ * is generated. This new population along with its predecessor is sorted using
+ * non-domination as the metric. Following this, the population is further
+ * segregated in fronts. A new population is generated from these fronts having
+ * size equal to that of the starting population.
+ *
+ * During evolution, two parents are randomly chosen using binary tournament
+ * selection. A pair of children are generated by crossing over these two
+ * candidates followed by mutation.
+ *
+ * The best front (Pareto optimal) is returned by the Optimize() method.
+ *
+ * For more information, see the following:
+ *
+ * @code
+ * @inproceedings{panichella2019adaptive,
+ * title={An adaptive evolutionary algorithm based on non-euclidean geometry for many-objective optimization},
+ * author={Panichella, Annibale},
+ * booktitle={Proceedings of the genetic and evolutionary computation conference},
+ * pages={595--603},
+ * year={2019}
+ * }
+ * @endcode
+ *
+ */
+class AGEMOEA
```

Comment:
If the directory is named `age_moea`, should we name this `AGE_MOEA`?  Or should we change the directory name to `agemoea/`?  I'm just looking for consistency, I don't have an opinion either way 👍